pinot-spi/src/main/java/org/apache/pinot/spi/data/readers/RecordExtractor.java

```diff
@@ -18,16 +18,28 @@
  */
 package org.apache.pinot.spi.data.readers;
 
-import org.apache.pinot.spi.data.Schema;
+import java.util.List;
 
+
+/**
+ * Extracts fields from input records
+ * @param <T> The format of the input record
+ */
 public interface RecordExtractor<T> {
+
+  /**
+   * Initialize the record extractor with its config
+   */
+  default void init(RecordExtractorConfig recordExtractorConfig) {
+  }
+
   /**
-   * TODO Add text to this javadoc
+   * Extracts fields as listed in the sourceFieldNames from the given input record and sets them into the GenericRow
    *
-   * @param schema
-   * @param from
-   * @param to
-   * @return
+   * @param sourceFieldNames List of field names to extract from the provided input record
+   * @param from The input record
+   * @param to The output GenericRow
+   * @return The output GenericRow
    */
-  GenericRow extract(Schema schema, T from, GenericRow to);
+  GenericRow extract(List<String> sourceFieldNames, T from, GenericRow to);
```

Comment:
Nice to extractor interface decoupled from Schema

Reply:
This comment is not addressed

Reply:
Done